neural/run_workflow.py

```diff
@@ -270,17 +281,20 @@ def run_neural_discern(data_partition, dsettypes, bertmodel, config, options, wr
                         doc_id = docs_id[doc_indx].item()
                         if(doc_id in bert_proc_docs):
                             # due to GPU limit
-                            embed_sents = ReaderWriter.read_data(bert_proc_docs[doc_id])
-                            embed_sents = embed_sents.to(device)  # send to gpu device
+                            # TODO: refactor away need for `.replace('.pkl', '_torch.pkl')` in uncased (normal bert) run
+                            embed_sents = torch.load(bert_proc_docs[doc_id], map_location=device)
+                            # embed_sents = embed_sents.to(device)  # send to gpu device
                         else:
-                            embed_sents = bert_encoder(docs_batch[doc_indx], docs_attn_mask[doc_indx], docs_len[doc_indx].item())
+                            embed_sents = bert_encoder(docs_batch[doc_indx], docs_attn_mask[doc_indx],
+                                                       docs_len[doc_indx].item())
                             # add embedding to dict
                             embed_fpath = os.path.join(sents_embed_dir, '{}.pkl'.format(doc_id))
                             ReaderWriter.dump_data(embed_sents, embed_fpath)
```

Comment:
I think here we should call `ReaderWriter.dump_tensor` since we are reading using `torch.load` which could be substituted by `ReaderWriter.read_tensor`. We won't have problem in this current setup because we typically generate the embeddings before doing the training/testing etc.. But in case you did not have the embeddings and you are running it for the first time then you will run into issue because you save the embedding using `ReaderWriter.dump_data` that uses `pickle` but when you read them in line `285` you use `torch.load` which eventually will not read the tensors and you will get some integer

Reply:
Good catch. I updated both the `ReaderWriter.dump_data` and the `torch.load` to the new `ReaderWriter` tensor methods.